src/test/java/de/qabel/desktop/ui/contact/ContactControllerTest.java

```diff
@@ -107,6 +107,26 @@ public void importContactsTest() throws URISyntaxException, PersistenceException
 		assertNotNull(c.getEcPublicKey());
 	}
 
+	@Test
+	public void changeIdentityObserverTest() throws Exception {
+		getContact("One");
+		Contact contact = getContact("Two");
+		controller = getController();
+
+		assertEquals(1, controller.contactsFromRepo.getContacts().size());
+		assertEquals("Two", controller.contactsFromRepo.getByKeyIdentifier(contact.getKeyIdentifier()).getAlias());
+
+	}
+
+	private Contact getContact(String name) throws PersistenceException {
```

Comment:
the method name doesn't reflect it's intention. It's more like `createIdentityContactPairAndSelectThem` which implies that it should probably be two methods.